src/application/deployer.go

```diff
@@ -46,15 +50,47 @@ func StartDeploy(deployConfig types.DeployConfig) error {
 		return err
 	}
 
+	terraformFilePath := filepath.Join(deployConfig.TerraformDir, "main.tf")
+	prevTerraformFileContents, err := readTerraformFile(terraformFilePath)
+	if err != nil {
+		return err
+	}
+
 	deployConfig.Logger.Log("Generating Terraform files...")
 	err = terraform.GenerateFile(deployConfig, imagesMap)
 	if err != nil {
 		return err
 	}
 
+	if deployConfig.DeployServicesOnly {
+		err = checkTerraformFile(terraformFilePath, prevTerraformFileContents)
+		if err != nil {
+			return err
+		}
+	}
+
 	return deployApplication(deployConfig, imagesMap)
 }
 
+func readTerraformFile(terraformFilePath string) ([]byte, error) {
+	var err error
+	if _, err = os.Stat(terraformFilePath); !os.IsNotExist(err) {
+		return ioutil.ReadFile(terraformFilePath)
+	}
+	return []byte{}, err
+}
+
+func checkTerraformFile(terraformFilePath string, prevTerraformFileContents []byte) error {
+	generatedTerraformFileContents, err := ioutil.ReadFile(terraformFilePath)
+	if err != nil {
+		return err
+	}
+	if !bytes.Equal(generatedTerraformFileContents, prevTerraformFileContents) {
+		return errors.New("'terraform/main.tf' file has changed. Please deploy manually to review these changes")
+	}
+	return nil
+}
```

Comment:
Can these functions be helpers in the terraform package and live alongside `WriteTerraformFile`? They could take in `deployConfig` instead of `terraformFilePath`. That one "main.tf" could live in a single location.